novawallet/Modules/NetworkManagement/CustomNetwork/Presenter/CustomNetworkBasePresenter.swift

```diff
@@ -0,0 +1,272 @@
+    func provideChainIdViewModel() {
+        let inputViewModel = InputViewModel.createNotEmptyInputViewModel(
+            for: partialChainId ?? "",
+            placeholder: Constants.chainIdPlaceholder
+        )
+        view?.didReceiveChainId(viewModel: inputViewModel)
+    }
+    
+    func provideBlockExplorerURLViewModel() {
+        let inputViewModel = InputViewModel.createNotEmptyInputViewModel(
+            for: partialBlockExplorerURL ?? "",
+            required: false,
+            placeholder: Constants.blockExplorerPlaceholder
+        )
+        view?.didReceiveBlockExplorerUrl(viewModel: inputViewModel)
+    }
+    
+    func provideCoingeckoURLViewModel() {
+        let inputViewModel = InputViewModel.createNotEmptyInputViewModel(
+            for: partialCoingeckoURL ?? "",
+            required: false,
+            placeholder: Constants.coingeckoTemplate
+        )
+        view?.didReceiveCoingeckoUrl(viewModel: inputViewModel)
+    }
+    
+    func provideNetworkTypeViewModel() {
+        view?.didReceiveNetworkType(
+            chainType,
+            show: knownChain == nil
+        )
+    }
+    
+    // MARK: Setup
+    
+    func completeSetup() {
+        provideViewModel()
+        provideButtonViewModel(loading: false)
+    }
+}
+
+// MARK: CustomNetworkPresenterProtocol
+
+extension CustomNetworkBasePresenter: CustomNetworkPresenterProtocol {
+    func setup() {
+        completeSetup()
+    }
+    
+    func select(segment: ChainType?) {
+        guard let segment else { return }
+        
+        chainType = segment
+        
+        cleanPartialValues()
+        provideViewModel()
+    }
+
+    func handlePartial(url: String) {
+        partialURL = url
+        
+        provideButtonViewModel(loading: false)
+    }
+
+    func handlePartial(name: String) {
+        partialName = name
+        
+        provideButtonViewModel(loading: false)
+    }
+    
+    func handlePartial(currencySymbol: String) {
+        partialCurrencySymbol = currencySymbol
+        
+        provideButtonViewModel(loading: false)
+    }
+    
+    func handlePartial(blockExplorerURL: String) {
+        partialBlockExplorerURL = blockExplorerURL
+        
+        provideButtonViewModel(loading: false)
+    }
+    
+    func handlePartial(coingeckoURL: String) {
+        partialCoingeckoURL = coingeckoURL
+        
+        provideButtonViewModel(loading: false)
+    }
+
+    func confirm() {
+        provideButtonViewModel(loading: true)
+        
+        actionConfirm()
+    }
+}
+
+// MARK: CustomNetworkBaseInteractorOutputProtocol
+
+extension CustomNetworkBasePresenter: CustomNetworkBaseInteractorOutputProtocol {
+    func didReceive(_ error: CustomNetworkBaseInteractorError) {
+        let close = R.string.localizable.commonClose(
+            preferredLanguages: selectedLocale.rLanguages
+        )
+        
+        let errorContent = error.toErrorContent(for: selectedLocale)
+        
+        wireframe.present(
```

Comment:
I think wireframe.present(error:...) is a shortcut for the error content presentable errors